_tutorials/new-component/create-a-new-component-part-5.md

```diff
@@ -0,0 +1,300 @@
+
+<pre><code class="o-syntax-highlight--js">	/**
+	 * Class constructor.
+	 * @param {HTMLElement} [exampleEl] - The component element in the DOM
+	 * @param {Object} [options={}] - An options object for configuring the component
+	 */
+	constructor (exampleEl, opts) {
+		this.exampleEl = exampleEl;
+		this.options = Object.assign({}, {
+		}, opts || Example.getDataAttributes(exampleEl));
+		// A property to store the current count.
+		this.count = 0;
+		// Listen to all click events on the o-example instance.
+		this.exampleEl.addEventListener('click', this);
+	}
+
+	/**
+	 * A method to handle event listeners.
+	 * https://medium.com/@WebReflection/dom-handleevent-a-cross-platform-standard-since-year-2000-5bf17287fd38
+	 * https://dom.spec.whatwg.org/#dom-eventlistener-handleevent
+	 * @param {Event} event
+	 */
+	handleEvent(event) {
+		// When any button within the `o-example` component is clicked
+		// increment the count.
+		if (event.target.tagName === 'BUTTON') {
+			this.count++;
+			// Log the count temporarily so we can see this working.
+			console.log(this.count);
+		}
+	}</code></pre>
+
+<figure>
+	<img alt="" src="/assets/images/tutorial-new-component/hello-world-demo-12-js.png" />
+	<figcaption class="o-typography-caption">
+        The `o-example` component with the browsers developer tools open. The numbers 1 to 5 have been logged as the button has been clicked five times.
+	</figcaption>
+</figure>
+
+Next we want to update our component to display the current count instead of logging the count to the browsers developer console. One way we could do that is my adding a `<span>` element to contain the current count and update its text content when the button is clicked. We'll identify the current count span with a namespaced data attribute `data-o-example-current-count`.
+
+<pre><code class="o-syntax-highlight--diff">
+&lt;div class="o-example &#123;&#123;#theme}}o-example--&#123;&#123;theme}}&#123;&#123;/theme}}" data-o-component="o-example">
+-    Hello world, I am a component named o-example!
++    Hello world, I am a component named o-example! You have clicked this lovely button &lt;span data-o-example-current-count>0&lt;/span> times.
+    &lt;button class="o-example__button">count&lt;/button>
+&lt;/div>
+</code></pre>
+
+Now in our JavaScript we can get any current count element and update it when our click count is incremented.
+
+<pre><code class="o-syntax-highlight--js">handleEvent(event) {
+    // When any button within the `o-example` component is clicked
+    // increment the count.
+    if (event.target.tagName === 'BUTTON') {
+        this.count++;
+        // Get all the elements within o-example with
+        // the attribute data-o-example-current-count.
+        const countElements = this.exampleEl.querySelectorAll('[data-o-example-current-count]');
+        // For each count element found, update the count.
+        countElements.forEach(e => e.innerText = this.count);
```

Comment:
ESLint was marking this line with an error because the forEach function was returning the value and forEach does not use the return value.

What do we think about changing this to a for-of loop instead of a forEach?
```suggestion
		for (const element of countElements) {
			element.innerText = this.count;
		}
```